### PR TITLE
4 - Vni phase 2 shares enhancement

### PR DESCRIPTION
**Pull request by deepaksibm**

### CODE REVIEW

File: website/docs/r/is_share.html.markdown [hunk 1]
```diff
@@ -70,13 +70,27 @@ The following arguments are supported:
 
     Nested scheme for `virtual_network_interface`:
     - `name` - (Required, String) Name for this virtual network interface.
+    - `id` - (Optional) The ID for virtual network interface. Mutually exclusive with other `virtual_network_interface` arguments.
+    - `allow_ip_spoofing` - (Optional, Bool) Indicates whether source IP spoofing is allowed on this interface. If false, source IP spoofing is prevented on this interface. If true, source IP spoofing is allowed on this interface.
+    - `auto_delete` - (Optional, Bool) Indicates whether this virtual network interface will be automatically deleted when target is deleted
+    - `enable_infrastructure_nat` - (Optional, Bool) If `true`:- The VPC infrastructure performs any needed NAT operations.- `floating_ips` must not have more than one floating IP.If `false`:- Packets are passed unchanged to/from the network interface,  allowing the workload to perform any needed NAT operations.- `allow_ip_spoofing` must be `false`.- If the virtual network interface is attached:  - The target `resource_type` must be `bare_metal_server_network_attachment`.  - The target `interface_type` must not be `hipersocket`.
```
Comment (astha-jain) [inline website/docs/r/is_share.html.markdown:76]: formatting and missing full example for create?

File: website/docs/r/is_share.html.markdown [hunk 2]
```diff
@@ -70,13 +70,27 @@ The following arguments are supported:
...
     - `primary_ip` - (Optional, List) The primary IP address to bind to the virtual network interface. May be either a reserved IP identity, or a reserved IP prototype object which will be used to create a new reserved IP.
 
         Nested scheme for `primary_ip`:
         - `auto_delete` - (Optional, Bool) Indicates whether this reserved IP member will be automatically deleted when either target is deleted, or the reserved IP is unbound. Defaults to `true`
         - `address` - (Optional, Forces new resource, String) The IP address to reserve. If unspecified, an available address on the subnet will automatically be selected.
         - `name`- (Optional, String) The name for this reserved IP. The name must not be used by another reserved IP in the subnet. Names starting with ibm- are reserved for provider-owned resources, and are not allowed.
         - `reserved_ip`- (Optional, String) The unique identifier for this reserved IP
+    - `ips` - (Optional, List) Additional IP addresses to bind to the virtual network interface.  May be either a reserved IP identity, or a reserved IP prototype object which will be used to create a new reserved IP.
```
Comment (astha-jain) [inline website/docs/r/is_share.html.markdown:84]: I thought we were not adding unsupported attributes to shares ?

File: website/docs/r/is_share_mount_target.html.markdown [hunk 3]
```diff
@@ -90,6 +90,25 @@ resource "ibm_is_share_mount_target" "mtarget1" {
...
+//Create mount target with VNI ID
+resource "ibm_is_subnet" "example" {
+  name                     = "my-subnet"
+  vpc                      = ibm_is_vpc.vpc2.id
+  zone                     = "br-sao-2"
+  total_ipv4_address_count = 16
+}
+resource "ibm_is_virtual_network_interface" "example"{
+	name = "my-example-vni"
+  subnet = ibm_is_subnet.example.id
```
Comment (astha-jain) [inline website/docs/r/is_share_mount_target.html.markdown:103]: fix formatting

File: website/docs/r/is_share_mount_target.html.markdown [hunk 4]
```diff
@@ -126,7 +151,9 @@ The following arguments are supported:
 
 The following attributes are exported:
 
-
+- `allow_ip_spoofing` - (Bool) Indicates whether source IP spoofing is allowed on this interface. If false, source IP spoofing is prevented on this interface. If true, source IP spoofing is allowed on this interface.
+  - `auto_delete` - (Bool) Indicates whether this virtual network interface will be automatically deleted when target is deleted
+  - `enable_infrastructure_nat` - (Bool) If `true`:- The VPC infrastructure performs any needed NAT operations.- `floating_ips` must not have more than one floating IP.If `false`:- Packets are passed unchanged to/from the network interface,  allowing the workload to perform any needed NAT operations.- `allow_ip_spoofing` must be `false`.- If the virtual network interface is attached:  - The target `resource_type` must be `bare_metal_server_network_attachment`.  - The target `interface_type` must not be `hipersocket`.
```
Comment (astha-jain) [inline website/docs/r/is_share_mount_target.html.markdown:156]: data source documentation?

File: ibm/service/vpc/resource_ibm_is_share.go [hunk 5]
```diff
@@ -165,6 +166,22 @@ func ResourceIbmIsShare() *schema.Resource {
...
+										Type:        schema.TypeBool,
+										Optional:    true,
+										Computed:    true,
+										Description: "Indicates whether this virtual network interface will be automatically deleted when`target` is deleted.",
+									},
+									"enable_infrastructure_nat": &schema.Schema{
+										Type:        schema.TypeBool,
+										Computed:    true,
+										Description: "If `true`:- The VPC infrastructure performs any needed NAT operations.- `floating_ips` must not have more than one floating IP.If `false`:- Packets are passed unchanged to/from the network interface,  allowing the workload to perform any needed NAT operations.- `allow_ip_spoofing` must be `false`.- If the virtual network interface is attached:  - The target `resource_type` must be `bare_metal_server_network_attachment`.  - The target `interface_type` must not be `hipersocket`.",
+									},
```
Comment (astha-jain) [inline ibm/service/vpc/resource_ibm_is_share.go:184]: tests?

File: website/docs/r/is_share_mount_target.html.markdown (hunk 4 above)
Comment (deepaksibm) [reply website/docs/r/is_share_mount_target.html.markdown:156]: share mount target GET response has only virtual_network_interface reference which is already added.
nothing new in response of mount target for Phase 2

File: website/docs/r/is_share.html.markdown (hunk 2 above)
Comment (deepaksibm) [reply website/docs/r/is_share.html.markdown]: removed from doc

File: website/docs/r/is_share_mount_target.html.markdown (hunk 3 above)
Comment (deepaksibm) [reply website/docs/r/is_share_mount_target.html.markdown:103]: fixed

File: ibm/service/vpc/resource_ibm_is_share.go (hunk 5 above)
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_share.go:184]: added

File: website/docs/r/is_share_mount_target.html.markdown (hunk 4 above)
Comment (deepaksibm) [reply website/docs/r/is_share_mount_target.html.markdown:156]: added some missing items

**Review by sumitkumartiwari — APPROVED** [review]

**Pull request merged**